python/lsst/ts/observatory/control/base_tcs.py

```diff
@@ -273,38 +260,75 @@ async def slew_icrs(
 
         alt_az = radec_icrs.transform_to(coord_frame_altaz)
 
-        if rottype == RotType.Sky:
-            self.log.debug(f"Setting sky angle to {rot}.")
-        elif rottype == RotType.Physical_Sky:
+        rot_frame = self.RotFrame.TARGET
+
+        # compute rotator physical position if rot_angle is sky.
+        rot_phys_val = salobj.angle_wrap_center(
+            Angle(
+                Angle(180.0, unit=u.deg)
+                + par_angle
+                - rot_angle
+                - (
+                    alt_az.alt
+                    if self.instrument_focus == InstrumentFocus.Nasmyth
+                    else 0.0
+                ),
+                unit=u.deg,
+            )
+        )
 
-            self.log.debug(f"Setting rotator physical position to {rot}.")
+        if rot_type == RotType.Sky:
+            self.log.debug(f"RotSky = {rot_angle}, RotPhys = {rot_phys_val}.")
+        elif rot_type == RotType.SkyAuto:
+            self.log.debug(f"Auto sky angle: {rot_angle}")
+            if not (self.rotator_limits[0] < rot_phys_val.deg < self.rotator_limits[1]):
+                self.log.debug(
+                    f"Rotator angle out of limits {rot_angle} [{self.rotator_limits}]. Wrapping."
+                )
+                rot_angle = salobj.angle_wrap_center(
+                    Angle(180.0, unit=u.deg) + rot_angle
+                )
```

Comment:
I'll have to consider that... I though about that in the beginning but I don't know if we really want to wander that off. The Scheduler originally had this strategy and it would only try to flip 180. Because that can impact the survey. Anyway, I'll think about it.